### PR TITLE
Add comparison info for Proton C

### LINKED ISSUE #6160: Improve marketing on proton-c page
This is the headline for the proton-c:

> The Proton C is an ARM STM32F303xC based drop-in replacement for the Pro Micro. It uses the same chip as the Planck rev6 boards, and Skull’s new ARM Clueboards.

While everything in there is probably correct (I didn't checked but I trust you) it says nothing to the average user. Even for mid users like me (I don't consider myself an advanced user, but I have several keyboards and I have even built my own from scratch) it does not tells me what I want to know. This is what I want to read first when I open that page:

- Does it have more memory? the answer seems to be YES! This is key!
- Does it have a faster processor? Again seems that the answer is YES!

And just with those two bullet points I'm sold. But currently, after reading that page I just feel: should I consider using this? Why?
I think we should make clearer that it is better and why it is better.

Regards

### PR DESCRIPTION
**Pull request by drashna**
Closes https://github.com/qmk/qmk_firmware/issues/6160

### CODE REVIEW

File: _pages/proton-c.md [hunk 1]
```diff
@@ -26,10 +26,27 @@ You can find the Proton C for purchase at:
 
 Support for Split Keyboards, RGB Underglow, and Backlighting are being worked on and should be available soon.
 
+## Comparison
+
+If you want to know why you should chose the Proton C for your next product, here is a comparison of the chips:
+
+---- | ATMega32u4        | AT90USB1286 | STM32F303xC
```
Comment (noroadsleft) [inline _pages/proton-c.md]: Do you even visual alignment, bro?

```markdown
 &nbsp;     | ATMega32u4       | AT90USB1286          | STM32F303xC
----------- | ---------------- | -------------------- | ----------------
Speed       | 16MHz            | 16MHz                | 72MHz
Voltage     | 5v               | 5v                   | 3.3v with some 5v capable pins
Flash size  | 32KB (28 usable) | 128KB (120KB usable) | 256KB (248KB usable)
EEPROM size | 1024 bytes       | 4096 Bytes           | 4096 bytes (emulated in flash)
Memory      | 2.5KB            | 8KB                  | 48KB
Timers      | 4 total          | 4 total              | 10 total
Digital IO  | 25 pins          | 46 pins              | 37 pins
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
For Line 9:

```markdown
The Proton C is an ARM STM32F303xC based drop-in replacement for the Pro Micro.  It uses the same chip as the Planck rev6 boards, and Skully's new ARM Clueboards.
```

File: _pages/proton-c.md (hunk 1 above)
Comment (drashna) [reply _pages/proton-c.md]: Nope, didn't bother.  GitHub will render it just fine. :D

**Comment by noroadsleft** [comment]
Line 9 still says `Skull's new ARM Clueboards`, but other than that:

:slightly_smiling_face:

**Review by jackhumbert — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]
:rofl:

**Pull request merged**